specs/superchain-upgrades.md

```diff
@@ -0,0 +1,199 @@
+
+A Superchain upgrade requires the node software to support up to a given Protocol Version.
+The version indicates support, the upgrade indicates the activation of new functionality.
+
+This document lists the protocol versions of the OP-Stack, starting at the Bedrock upgrade,
+as well as the default Superchain Targets.
+
+Activation rule parameters of network upgrades are configured as part of the Superchain Target specification:
+chains following the same Superchain Target upgrade synchronously.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Protocol Version](#protocol-version)
+  - [Protocol Version Format](#protocol-version-format)
+    - [Major versions](#major-versions)
+    - [Minor versions](#minor-versions)
+    - [Patch versions](#patch-versions)
+  - [Protocol Version Exposure](#protocol-version-exposure)
+- [Superchain Target](#superchain-target)
+  - [Superchain Version signaling](#superchain-version-signaling)
+- [Activation rules](#activation-rules)
+  - [L2 Block-number based activation (deprecated)](#l2-block-number-based-activation-deprecated)
+  - [L2 Block-timestamp based activation](#l2-block-timestamp-based-activation)
+- [Post-Bedrock Network upgrades](#post-bedrock-network-upgrades)
+  - [Regolith](#regolith)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Protocol Version
+
+The Protocol Version documents the progression of the total set of canonical OP-Stack specifications.
+Components of the OP-Stack implement the subset of their respective protocol component domain,
+up to a given Protocol Version of the OP-Stack.
+
+OP-Stack mods, i.e. non-canonical extensions to the OP-Stack, are not included in the versioning of the Protocol.
+Instead, mods must specify which upstream Protocol Version they are based on and where breaking changes are made.
+This ensures tooling of the OP-Stack can be shared and collaborated on with OP-Stack mods.
+
+The Protocol Version is NOT a hardfork identifier, but rather indicates software-support for a well-defined set
+of features introduced in past and future hardforks, not the activation of said hardforks.
+
+Changes that can be included in prospective Protocol Versions may be included in the specifications as proposals,
+with explicit notice of the Protocol Version they are based on.
+This enables an iterative integration process into the canonical set of specifications,
+but does not guarantee the proposed specifications become canonical.
+
+Note that the Protocol Version only applies to the Protocol specifications with the Superchain Targets specified within.
+This versioning is independent of the [Semver] versioning used in OP Stack smart-contracts,
+and the [Semver]-versioned reference software of the OP-Stack.
+
+### Protocol Version Format
+
+The Protocol Version is [Semver]-compatible.
```

Comment:
Now that we use sermver-compatible protocol versions instead of a simple integer, do we want to recommend different behavior of node implementations when e.g. the major vs minor version number is increased? For example, that we recommend halting the node if the supported major version is lower than the required, but not necessarily recommend this for minor version bumps.